pkg/internal/flags.go

```diff
@@ -0,0 +1,20 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
```

Comment:
Note for other reviewers, this package is at the right level because we'll probably use the same variable for OpenAPI v3 that lives in `pkg/spec3`. Thanks!

Reply:
Should this be protected by a build tag rather than a variable?

I've generally recommended against open source projects using `go-json-experiment` due to the following effect:
1. Let module M depend on modules A and B.
2. Module A depends on `go-json-experiment/json@v0.5.0`.
3. Module B depends on `go-json-experiment/json@v0.8.0`.
4. Let’s suppose a breaking change occurs between `v0.5.0` and `v0.8.0`.
5. [MVS](https://research.swtch.com/vgo-mvs) dictates that `v0.8.0` be selected to build module M.
6. However, the use of `v0.8.0` can break package A since it is using the API for `v0.5.0`, which is not compatible.

I don't know enough about the `kubernetes/kube-openapi` whether this is a module that is only used to build a binary, or whether it is a module intended to be depended upon by many other modules. If the former, then guarding it with a variable sounds fine. If the latter, then we probably want to guard it with a build tag.

Reply:
@liggitt Do you have an opinion about the suggestion above? Thanks.

Reply:
Forking SGTM.

The license is BSD and the listed authors are the Go authors since I had originally started the project when I was a Google employee. It is currently developed by all non-Googlers, but every contributor is also a regular contributor to the Go toolchain, so have each signed the Google CLA.

Reply:
forking into an `internal` `third_party` package with correct attribution as an internal implementation detail until the go-json-experiment stabilizes sgtm

Reply:
I've forked it into `pkg/internal/third_party/go-json-experiment/json`. I added a readme with the commit hash and a note about when it should be removed. 

To fork I executed `git clone` and removed `.git`, `.github`, `go.mod`, `go.sum`, and all `.png` files. I am unsure if there is a more standard way to "fork" the package into our tree.